src/GitHub.VisualStudio.UI/Helpers/ThemeDictionaryManager.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Windows;
+using System.ComponentModel;
+using Microsoft.VisualStudio.PlatformUI;
+using GitHub.VisualStudio.Helpers;
+using GitHub.Helpers;
+
+namespace GitHub.VisualStudio.UI.Helpers
+{
+    public class ThemeDictionaryManager : SharedDictionaryManagerBase, IDisposable
+    {
+        static bool isInDesignMode;
+        Uri baseThemeUri;
+
+        static ThemeDictionaryManager()
+        {
+            isInDesignMode = DesignerProperties.GetIsInDesignMode(new DependencyObject());
+        }
+
+        public override Uri Source
+        {
+            get { return base.Source; }
+            set
+            {
+                InitTheme(value);
+                base.Source = GetCurrentThemeUri();
+            }
+        }
+
+        public void Dispose()
+        {
+            if(baseThemeUri != null)
+            {
+                baseThemeUri = null;
+                if (!isInDesignMode)
+                {
+                    VSColorTheme.ThemeChanged -= OnThemeChange;
+                }
+            }
+        }
+
+        void InitTheme(Uri themeUri)
```

Comment:
You know what would be nice? Being able to change the theme in the designer. We could expose a `ThemeName` property or similar on this class which would allow you to toggle the theme at design time and would have no effect at runtime. This should probably go in a different PR though ;)